pkg/agent/nodeportlocal/npl_agent_test.go

```diff
@@ -388,12 +423,16 @@ func TestPodDelete(t *testing.T) {
 	assert.NoError(t, err, "Error when polling for port table update")
 }
 
-// TestPodAddMultiPort creates a Pod with multiple ports and verifies that the Pod's NPL annotation
-// and the local port table are updated correctly.
-func TestPodAddMultiPort(t *testing.T) {
-	testSvc := getTestSvc()
-	testPod := getTestPod()
+// TestPodAddMultiPort creates a Pod with multiple ports and a Service with same ports as target ports.
+// It is verified that the Pod's NPL annotationthe local port table are updated correctly.
+func TestAddMultiPortPodSvc(t *testing.T) {
 	newPort := 90
+	testSvc := getTestSvc(defaultPort, int32(newPort))
+	testPod := getTestPod()
+	testPod.Spec.Containers[0].Ports = append(
+		testPod.Spec.Containers[0].Ports,
+		corev1.ContainerPort{ContainerPort: defaultPort},
+	)
 	testPod.Spec.Containers[0].Ports = append(
 		testPod.Spec.Containers[0].Ports,
 		corev1.ContainerPort{ContainerPort: int32(newPort)},
```

Comment:
adding Pod container ports is not required any more (now that we use the Service's target ports), unless we are dealing with a named port. Maybe for this test case we should omit the container ports.

Reply:
@monotosh-avi ping on this, any thoughts?

Reply:
missed this, updated now